src/GraphQL/Types/Schema.cs

```diff
@@ -411,8 +412,41 @@ protected virtual void Validate()
             //TODO: add different validations, also see SchemaBuilder.Validate
             //TODO: checks for parsed SDL may be expanded in the future, see https://github.com/graphql/graphql-spec/issues/653
             // Do not change the order of these validations.
+            CoerceInputTypeDefaultValues();
             SchemaValidationVisitor.Instance.Run(this);
             AppliedDirectivesValidationVisitor.Instance.Run(this);
         }
+
+        protected virtual void CoerceInputTypeDefaultValues()
+        {
+            var completed = new HashSet<IInputObjectGraphType>();
+            var inProcess = new Stack<IInputObjectGraphType>();
+            foreach (var type in AllTypes.Dictionary.Values)
+            {
+                if (type is IInputObjectGraphType inputType)
+                    ExamineType(inputType, completed, inProcess);
+            }
+
+            static void ExamineType(IInputObjectGraphType inputType, HashSet<IInputObjectGraphType> completed, Stack<IInputObjectGraphType> inProcess)
```

Comment:
Is there a speed difference between this and:

```suggestion
            void ExamineType(IInputObjectGraphType inputType)
```

?  If the compiler produces the same output either way, then the suggestion would be cleaner.

Reply:
If compiler in both ways generate static method with struct arg then I would prefer this suggestion. I think that explicit `static` keyword for local functions helps to see unintentional arguments closure.

Reply:
I don't know if it does or not, so I'll leave the code as-is.  Feel free to merge the suggestion if you think it would be better the other way.